### PR TITLE
[Improvement][Task Plugin]Add Basic Authentication for HttpTask

### PR DESCRIPTION
**Pull request by LuXugang**
Basic Authentication is a common security mechanisms, we should at least have it.

### CODE REVIEW

File: dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/main/java/org/apache/dolphinscheduler/plugin/task/http/HttpTask.java [hunk 1]
```diff
@@ -289,8 +293,20 @@ protected void setHeaders(HttpUriRequest request, List<HttpProperty> httpPropert
      */
     protected CloseableHttpClient createHttpClient() {
         final RequestConfig requestConfig = requestConfig();
-        HttpClientBuilder httpClientBuilder;
-        httpClientBuilder = HttpClients.custom().setDefaultRequestConfig(requestConfig);
+        HttpClientBuilder httpClientBuilder = HttpClients.custom();
+        if (httpParameters.getEnableProxy()) {
+            CredentialsProvider provider = new BasicCredentialsProvider();
+            AuthScope authScope = null;
```
Comment (LuXugang) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/main/java/org/apache/dolphinscheduler/plugin/task/http/HttpTask.java:299]: Basic Authentication does not always need proxy, just username and password could be the least parameter

File: dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/main/java/org/apache/dolphinscheduler/plugin/task/http/HttpParameters.java [hunk 2]
```diff
@@ -132,4 +142,44 @@ public int getSocketTimeout() {
...
+    public String getUser() {
+        return user;
+    }
+
+    public void setUser(String user) {
+        this.user = user;
+    }
+
+    public String getPassword() {
+        return password;
```
Comment (ruanwenjun) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/main/java/org/apache/dolphinscheduler/plugin/task/http/HttpParameters.java]: Please use Lombok to generate these get/set method.

Comment (LuXugang) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/main/java/org/apache/dolphinscheduler/plugin/task/http/HttpParameters.java]: Address in [bae7d65](https://github.com/apache/dolphinscheduler/pull/14491/commits/bae7d65d7b2cca77fd7f0f658e6205c6ac087eec)

**Review by Radeity — COMMENTED** [review]
You should modify front-end and doc, either.

File: dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/main/java/org/apache/dolphinscheduler/plugin/task/http/HttpTask.java [hunk 3]
```diff
@@ -289,8 +293,20 @@ protected void setHeaders(HttpUriRequest request, List<HttpProperty> httpPropert
...
+            // basic auth could tak effect alone without proxy
```
Comment (Radeity) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/main/java/org/apache/dolphinscheduler/plugin/task/http/HttpTask.java]: Typo error here -> `tak`

Comment (LuXugang) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/main/java/org/apache/dolphinscheduler/plugin/task/http/HttpTask.java]: Address in [f1ce0d8](https://github.com/apache/dolphinscheduler/pull/14491/commits/f1ce0d8e1024727a3cb36b97932c4a2a596ad682) .

**Comment by LuXugang** [comment]
> You should modify front-end and doc, either.

Hi @Radeity,  could front-end's code proposed in another PR by my colleague. I am not good at front-end stack.

**Comment by Radeity** [comment]
> Hi @Radeity, could front-end's code proposed in another PR by my colleague. I am not good at front-end stack.

No problem, your should better create an issue about it, in which can let us know your colleague will follow up the front-end part. In addition, you can add doc first :D

**Comment by ruanwenjun** [comment]
> > You should modify front-end and doc, either.
> 
> Hi @Radeity, could front-end's code proposed in another PR by my colleague. I am not good at front-end stack.

Could you please let your colleague work on this PR, it's not a good practice split one feature at several PR.

**Comment by LuXugang** [comment]
> Could you please let your colleague work on this PR, it's not a good practice split one feature at several PR

addressed in e2c69b7

**Comment by Radeity** [comment]
Hi, @LuXugang , please fix CI errors.

**Review by songjianet — APPROVED** [review]

**Pull request closed without merging**